src/Bicep.Cli.IntegrationTests/ProgramTests.cs

```diff
@@ -108,84 +115,99 @@ public void ZeroFilesToBuildShouldProduceError()
         [DynamicData(nameof(GetValidDataSets), DynamicDataSourceType.Method, DynamicDataDisplayNameDeclaringType = typeof(DataSet), DynamicDataDisplayName = nameof(DataSet.GetDisplayName))]
         public void BuildSingleFileShouldProduceExpectedTemplate(DataSet dataSet)
         {
-            var bicepFilePath = FileHelper.SaveResultFile(this.TestContext!, Path.Combine(dataSet.Name, DataSet.TestFileMain), dataSet.Bicep);
+            var outputDirectory = dataSet.SaveFilesToTestDirectory(TestContext, dataSet.Name);
+            var bicepFilePath = Path.Combine(outputDirectory, DataSet.TestFileMain);
 
-            var (output, error) = TextWriterHelper.InvokeWriterAction((@out, err) =>
+            var (output, error, result) = TextWriterHelper.InvokeWriterAction((@out, err) =>
             {
                 var p = CreateProgram(@out, err);
-                p.Run(new[] { "build", bicepFilePath }).Should().Be(0);
+                return p.Run(new[] { "build", bicepFilePath });
             });
 
-            output.Should().BeEmpty();
-            error.Should().BeEmpty();
+            using (new AssertionScope())
```

Comment:
This is cool. I need to use it a lot more.

Reply:
Yup, I was banging my head against the "0 should be 1" assertions - this is a nice solution for including extra context.